vec3d.h

```diff
@@ -0,0 +1,130 @@
+#include<math.h>
+
+
+template<typename T>
```

Comment:
Another possible improvement here is by using C++20 Concepts to constrain `T` to `float` and `double`.